Merge Conflict/Assets/Scripts/Menu/UpgradeMenu.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        mergeXPButton.onClick.AddListener(() => UpgradeMergeXP(Upgrades.MergeXPUpgrade));
+        moneyWhenTrashedButton.onClick.AddListener(() => UpgradeMoneyWhenTrashed(Upgrades.MoneyWhenTrashedUpgrade));
+        spawnChanceWhenTrashDiscardedButton.onClick.AddListener(() => UpgradeSpawnChanceWhenTrashDiscarded(Upgrades.SpawnChanceWhenTrashDiscardedUpgrade));
+        spawnIntervalButton.onClick.AddListener(() => UpgradeSpawnInterval(Upgrades.SpawnIntervalUpgrade));
+
+        // TODO: delete this test money function later
+#if UNITY_EDITOR
+        MoneyHandler.Instance.AddMoney(100000);
+#endif
+
+        UpdateUI();
+    }
+
+    void Update()
+    {
+    }
+
+    public void UpdateUI()
+    {
+        levelText.text = "Level " + ExperienceSystem.ExperienceHandler.GetCurrentLevel().ToString();
+        moneyText.text = MoneyHandler.Instance.Money.ToString() + " $";
+
+        UpdateDescriptionsTexts();
+        UpdateButtons();
+    }
+
+
+    void UpdateButtons()
+    {
+        // Check if next level is available
+        mergeXPButtonText.text = Upgrades.MergeXPUpgrade.isAtMaxLevel() ? "Max" : $"{Upgrades.MergeXPUpgrade.CostForNextLevel[Upgrades.MergeXPUpgrade.Level]} $";
+        moneyWhenTrashedButtonText.text = Upgrades.MoneyWhenTrashedUpgrade.isAtMaxLevel() ? "Max" : $"{Upgrades.MoneyWhenTrashedUpgrade.CostForNextLevel[Upgrades.MoneyWhenTrashedUpgrade.Level]} $";
+        spawnChanceWhenTrashDiscardedButtonText.text = Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.isAtMaxLevel() ? "Max" : $" {Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.CostForNextLevel[Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.Level]} $";
+        spawnIntervalButtonText.text = Upgrades.SpawnIntervalUpgrade.isAtMaxLevel() ? "Max" : $"{Upgrades.SpawnIntervalUpgrade.CostForNextLevel[Upgrades.SpawnIntervalUpgrade.Level]} $";
+    }
+
+    void UpdateDescriptionsTexts()
+    {
+        mergeXPDescriptionText.text = $"{Upgrades.MergeXPUpgrade.GetCurrentLevelDescription()}" + (!Upgrades.MergeXPUpgrade.isAtMaxLevel() ? $"\n -> Next Level ({Upgrades.MergeXPUpgrade.GetNextLevelDescription()})" : "");
+        moneyWhenTrashedDescriptionText.text = $"{Upgrades.MoneyWhenTrashedUpgrade.GetCurrentLevelDescription()}" + (!Upgrades.MoneyWhenTrashedUpgrade.isAtMaxLevel() ? $"\n -> Next Level ({Upgrades.MoneyWhenTrashedUpgrade.GetNextLevelDescription()})" : "");
+        spawnChanceWhenTrashDiscardedDescriptionText.text = $"{Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.GetCurrentLevelDescription()}" + (!Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.isAtMaxLevel() ? $"\n -> Next Level ({Upgrades.SpawnChanceWhenTrashDiscardedUpgrade.GetNextLevelDescription()})" : "");
+        spawnIntervalDescriptionText.text = $"{Upgrades.SpawnIntervalUpgrade.GetCurrentLevelDescription()}" + (!Upgrades.SpawnIntervalUpgrade.isAtMaxLevel() ? $"\n -> Next Level ({Upgrades.SpawnIntervalUpgrade.GetNextLevelDescription()})" : "");
+    }
+
+    void UpgradeMergeXP(MergeXPUpgrade upgrade)
```

Comment:
All Upgrade... Functions are functionally the same, only the Debug Message is different. These 4 functions, should be a single function. 
If the Debug Message is really necessary, then i would implement a function in Upgrade class, that each upgrade class could overwrite.

Reply:
The debug message is not necessary and will be deleted.

Reply:
Function "BuyNextUpgradeLevel()" is implemented now.